components/html-block/html-block.js

```diff
@@ -274,20 +293,14 @@ class HtmlBlock extends RtlMixin(LitElement) {
 	}
 
 	_stamp(slot) {
-		const stampHTML = async nodes => {
-			if (nodes && nodes.length > 0) {
-
-				let temp = document.createElement('div');
-				temp.style.display = 'none';
-				nodes.forEach(node => temp.appendChild(node.cloneNode(true)));
+		const renderContainer = this.shadowRoot.querySelector('.d2l-html-block-rendered');
 
-				this._renderContainer.appendChild(temp);
-				temp = await this._processRenderers(temp);
-				this._renderContainer.innerHTML = temp.innerHTML;
+		const stampHTML = async nodes => {
+			renderContainer.innerHTML = '';
+			if (!nodes || nodes.length === 0) return;
 
-			} else {
-				this._renderContainer.innerHTML = '';
-			}
+			nodes.forEach(node => renderContainer.appendChild(node.cloneNode(true)));
+			await this._processRenderers(renderContainer);
```

Comment:
It's important that we stamp the slot contents into the render container **before** we run any of the renderers. MathJax in particular requires that these be attached to the DOM or they won't typeset properly.

Reply:
Almost worth include a comment to that effect... `// must attach nodes before processing renderer plugins since some require connected nodes (ex. MathJax)"`.

Reply:
Good call, yeah, otherwise this requirement isn't at all obvious and someone could easily come along and switch the order without realizing it's going to subtly break stuff.